### PR TITLE
Move the logic of excluding contacts of DMs already included in reports to `filterOptions`

### PR DESCRIPTION
**Pull request by c3024**
### Details

Presently, `getOptions` excludes the contacts of DM reports included in `recentReports`. These are reduced to 5 recent reports in the participant selection page of "Submit Expense" flow if we pass a `maxRecentReportsToShow` to `filterOptions`. Because the contacts of these DM reports are excluded already in `getOptions`, these do not appear in the "Contacts" section. The logic of exclusion of Contacts for reports added in recent reports is moved from `getOptions` to `filterOptions` to fix this issue.

### Fixed Issues

$ https://github.com/Expensify/App/issues/48114
PROPOSAL: https://github.com/Expensify/App/issues/48114#issuecomment-2323418693

### Tests

[Pre-requisite]: Login an account with more than 5 (say 10 DM) reports.

Test 1:
1. Click on FAB > "Submit Expense" > "Manual" > Input any amount > "Next" 
2. Do not input anything in the search box yet
3. Verify that 
-- there are 5 recent reports in "Recents" section
-- all contacts of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM shown in "Recents" section is not shown in "Contacts" section
4. Input a search term
5. Verify that 
-- there are 5 recent reports matching with the search term in "Recents" section
-- all contacts (matching with the search term) of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM report matching with the search term shown in "Recents" section is not shown in "Contacts" section

Test 2:
1. Click on FAB > "Start Chat"
2. Follow steps 2 to 5 of Test 1 and verify the results

Test 3:
1. Go to any DM report
2. Click on "+" in composer > "Assign task"
3. Input title > "Next"
4. Follow steps 2 to 5 of Test 1 and verify the results

Test 4:
1. Click on "Settings" > "Workspaces" > Click on any workspace > "Members" > "Invite Member"
2. Verify that all "Contacts" are shown in options
3. Input a text in the search input
4. Verify that "Contacts" matching with the search term are shown in the options
 
- [x] Verify that no errors appear in the JS console

### Offline tests

[Pre-requisite]: Login an account with more than 5 (say 10 DM) reports.

Test 1:
1. Click on FAB > "Submit Expense" > "Manual" > Input any amount > "Next" 
2. Do not input anything in the search box yet
3. Verify that 
-- there are 5 recent reports in "Recents" section
-- all contacts of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM shown in "Recents" section is not shown in "Contacts" section
4. Input a search term
5. Verify that 
-- there are 5 recent reports matching with the search term in "Recents" section
-- all contacts (matching with the search term) of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM report matching with the search term shown in "Recents" section is not shown in "Contacts" section

Test 2:
1. Click on FAB > "Start Chat"
2. Follow steps 2 to 5 of Test 1 and verify the results

Test 3:
1. Go to any DM report
2. Click on "+" in composer > "Assign task"
3. Input title > "Next"
4. Follow steps 2 to 5 of Test 1 and verify the results

Test 4:
1. Click on "Settings" > "Workspaces" > Click on any workspace > "Members" > "Invite Member"
2. Verify that all "Contacts" are shown in options
3. Input a text in the search input
4. Verify that "Contacts" matching with the search term are shown in the options
 
### QA Steps

[Pre-requisite]: Login an account with more than 5 (say 10 DM) reports.

Test 1:
1. Click on FAB > "Submit Expense" > "Manual" > Input any amount > "Next" 
2. Do not input anything in the search box yet
3. Verify that 
-- there are 5 recent reports in "Recents" section
-- all contacts of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM shown in "Recents" section is not shown in "Contacts" section
4. Input a search term
5. Verify that 
-- there are 5 recent reports matching with the search term in "Recents" section
-- all contacts (matching with the search term) of DMs not included in the "Recents" section are included in the "Contacts" section
-- contact of any DM report matching with the search term shown in "Recents" section is not shown in "Contacts" section

Test 2:
1. Click on FAB > "Start Chat"
2. Follow steps 2 to 5 of Test 1 and verify the results

Test 3:
1. Go to any DM report
2. Click on "+" in composer > "Assign task"
3. Input title > "Next"
4. Follow steps 2 to 5 of Test 1 and verify the results

Test 4:
1. Click on "Settings" > "Workspaces" > Click on any workspace > "Members" > "Invite Member"
2. Verify that all "Contacts" are shown in options
3. Input a text in the search input
4. Verify that "Contacts" matching with the search term are shown in the options
 

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a8899fc7-a82d-4b74-8905-f29df523efc5

https://github.com/user-attachments/assets/999827a4-a4d3-4e36-b128-b517fcc64718

https://github.com/user-attachments/assets/7863412f-7c5d-4c2f-a3ae-e353350ee0ed

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/0d7ac3af-a2aa-437c-95be-e62c60bb9238

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/3ba1e6c8-f03a-41f4-ad58-a7acc60b8931

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/4821d30f-805f-4b1d-b391-b93d8f3182c7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/210f9753-546b-453f-b94f-f067c80b8cae

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/1c6e9367-16f7-453e-a207-555b52aaeafb

</details>

### CODE REVIEW

**Comment by eh2077** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by eh2077 — COMMENTED** [review]
Tested various cases and all look good. Just a comment worrying about possible performance downgrade.

File: src/libs/OptionsListUtils.ts [hunk 1]
```diff
@@ -2569,9 +2574,10 @@ function filterOptions(options: Options, searchInputValue: string, config?: Filt
     if (maxRecentReportsToShow > 0 && recentReports.length > maxRecentReportsToShow) {
         recentReports.splice(maxRecentReportsToShow);
     }
+    const filteredPersonalDetails = filteredPersonalDetailsOfRecentReports(recentReports, personalDetails);
```
[the remark this answers is not in the document]
Comment (c3024) [reply src/libs/OptionsListUtils.ts:2577]: It will but that might not be significant.

We iterate over all `recentReports` here for each search word.

https://github.com/Expensify/App/blob/9d86f30d94405fc184c4d0cb7d5cbcfd54b7fb82/src/libs/OptionsListUtils.ts#L2520-L2542

We iterate over the `recentReports` once more for this function `filteredPersonalDetailsOfRecentReports`.

Since, we pass `searchTerm` only to this function `filterOptions` and not `getOptions`, we need to do this exclusion of contacts only here.

**Review by eh2077 — APPROVED** [review]
LGTM

**Review by MonilBhavsar — COMMENTED** [review]
I am going to review it on Monday.
Should we add lint rule to prevent wrong usage of filterOptions() as we discussed?

**Comment by c3024** [comment]
>Should we add lint rule to prevent wrong usage of filterOptions() as we discussed?

I checked the `.eslintrc.js`. It does not have any such custom rules for functions. If we do this, this will be the first. So, I am wondering if there is a reason (perhaps complexity) for never using them till now.

We have two other options.
1. We can throw a console warning or error like this in `getOptions`

```js
if (options && options.reports && options.reports.length !== 0 && searchInputValue) {
        console.error('getOptions: Do not provide searchInputValue when options.reports is not empty. Use the performant filterOptions to filter with searchInputValue.');
    }
```

2. We can modify the types to enforce either non-empty `options` or non-empty `searchInputValue` can only be passed to `getOptions` but not both of them.

**Comment by MonilBhavsar** [comment]
> We can modify the types to enforce either non-empty options or non-empty searchInputValue can only be passed to getOptions but not both of them.

This sounds good, plus we can add a comment 👍 
@eh2077 curious for your thoughts too

**Comment by eh2077** [comment]
Yeah, I also agree - modifying types to achieve it sounds more viable 👍

**Comment by MonilBhavsar** [comment]
Thanks! Let's do it 👍

**Comment by c3024** [comment]
I have not found a way to enforce the constraint in `getOptions`. However, I could enforce the constraint in `getFilteredOptions`.

Here is the [branch](https://github.com/c3024/App/tree/enforce-empty-search-term-with-options) with some files changed.

When we pass a search term in cases with reports or personal details (in the `MoneyRequestParticipantsSelector` file in the video), TypeScript complains.

In cases where they are not passed, we can pass a search term, and TypeScript does not complain (as seen in `EditReportFieldDropdownPage` in the video).


https://github.com/user-attachments/assets/9201c926-8c4c-4d12-9b83-b3e679cf2e36


This type modification can also make the usage of `getFilteredOptions` cleaner everywhere. However, this requires careful changes in the params of `getFilteredOptions` used at different places.

I think we should handle that in a separate PR so that if we make any mistakes, we can revert it easily.

**Comment by eh2077** [comment]
@c3024 Thanks for the update! Is it possible to achieve what we want through function overloads? See https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

**Comment by c3024** [comment]
I did try with functional overloads with `getOptions`.  Since, we do not directly use `getOptions`, I could not make it work to correctly enforce the `searchInputValue` as the empty string literal `''`. By the time it passes from `getFilteredOptions` et al TypeScript starts treating it as a `string` and starts complaining.

File: src/libs/OptionsListUtils.ts [hunk 2]
```diff
@@ -2210,34 +2205,68 @@ function getIOUConfirmationOptionsFromPayeePersonalDetail(personalDetail: OnyxEn
...
+        includeTaxRates = false,
+        maxRecentReportsToShow = CONST.IOU.MAX_RECENT_REPORTS_TO_SHOW,
+        taxRates = {} as TaxRatesWithDefault,
+        includeSelfDM = false,
+        includePolicyReportFieldOptions = false,
+        policyReportFieldOptions = [],
+        recentlyUsedPolicyReportFieldOptions = [],
+        includeInvoiceRooms = false,
+        action,
+        sortByReportTypeInSearch,
```
Comment (eh2077) [inline src/libs/OptionsListUtils.ts:2268]: To keep it same as before, it should have default value `false` right?

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx [hunk 3]
```diff
@@ -114,40 +114,26 @@ function MoneyRequestParticipantsSelector({
...
-            [],
-            false,
-            {},
-            [],
+            includeP2P: (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
             // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-            (canUseP2PDistanceRequests || iouRequestType !== CONST.IOU.REQUEST_TYPE.DISTANCE) && !isCategorizeOrShareAction,
-            false,
-            false,
-            0,
```
Comment (eh2077) [inline src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:141]: `maxRecentReportsToShow: 0` is missing

**Review by eh2077 — COMMENTED** [review]
looks good, just minor comments

File: src/pages/iou/request/MoneyRequestParticipantsSelector.tsx (hunk 3 above)
Comment (eh2077) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:141]: @c3024 Btw this is crucial missing 😂

Comment (c3024) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:141]: Yea, my bad!

Comment (c3024) [reply src/pages/iou/request/MoneyRequestParticipantsSelector.tsx:141]: Done.


https://github.com/user-attachments/assets/9d9ae265-8856-4734-872c-b262111dfe72

**Review by eh2077 — APPROVED** [review]
Retested it on web and works well!

@c3024 Thanks for your patience!

File: tests/perf-test/OptionsListUtils.perf-test.ts [hunk 4]
```diff
@@ -105,9 +105,12 @@ describe('OptionsListUtils', () => {
     });
 
     /* Testing getFilteredOptions */
-    test('[OptionsListUtils] getFilteredOptions with search value', async () => {
```
Comment (MonilBhavsar) [inline tests/perf-test/OptionsListUtils.perf-test.ts:108]: We're still passing search value, then why this change?

File: tests/perf-test/OptionsListUtils.perf-test.ts [hunk 5]
```diff
@@ -105,9 +105,12 @@ describe('OptionsListUtils', () => {
...
+    test('[OptionsListUtils] getFilteredOptions', async () => {
         await waitForBatchedUpdates();
-        await measureFunction(() => OptionsListUtils.getFilteredOptions(options.reports, options.personalDetails, mockedBetas, SEARCH_VALUE));
+        // Ideally getFilteredOptions should not be used with both options and search value
+        // The more performant filterOptions should be used instead to pass search value with options containing reports and personal details
```
Comment (MonilBhavsar) [inline tests/perf-test/OptionsListUtils.perf-test.ts:111]: I think we need to rephrase this comment. 
```suggestion
        // It's recommended not to use getFilteredOptions with both options and a search value
        // For better performance, use filterOptions instead, especially when passing a search value with options that include reports and personal details.
```

File: tests/unit/OptionsListUtilsTest.ts [hunk 6]
```diff
@@ -2880,7 +2837,8 @@ describe('OptionsListUtils', () => {
         });
 
         it('should not return any options or user to invite if contact number contains alphabet characters (getFilteredOptions)', () => {
-            const options = OptionsListUtils.getFilteredOptions(OPTIONS.reports, OPTIONS.personalDetails, [], '');
+            // const options = OptionsListUtils.getFilteredOptions(OPTIONS.reports, OPTIONS.personalDetails, [], '');
```
Comment (MonilBhavsar) [inline tests/unit/OptionsListUtilsTest.ts:2840]:
```suggestion
```

File: tests/unit/OptionsListUtilsTest.ts [hunk 7]
```diff
@@ -2833,7 +2789,8 @@ describe('OptionsListUtils', () => {
         });
 
         it('should not return any options but should return an user to invite if no matching options exist and the search value is a potential email (getFilteredOptions)', () => {
-            const options = OptionsListUtils.getFilteredOptions(OPTIONS.reports, OPTIONS.personalDetails, [], '');
+            // const options = OptionsListUtils.getFilteredOptions(OPTIONS.reports, OPTIONS.personalDetails, [], '');
```
Comment (MonilBhavsar) [inline tests/unit/OptionsListUtilsTest.ts:2792]:
```suggestion
```

**Review by MonilBhavsar — COMMENTED** [review]
Thanks for TS cleanup. Looks good!

File: tests/perf-test/OptionsListUtils.perf-test.ts (hunk 4 above)
Comment (c3024) [reply tests/perf-test/OptionsListUtils.perf-test.ts:108]: Oh, initially I removed the search value from this test to make it consistent with the use of the function now but then changed because this is a performance test. So, I forgot to update the test name back. Will update.

**Comment by c3024** [comment]
@MonilBhavsar 

Thanks for the comments. Updated with these suggestions.

I added a couple of comments in the last commit explaining why we are enforcing the types. Please have a look!

**Review by MonilBhavsar — APPROVED** [review]
